utils/sample_qc.py

```diff
@@ -364,98 +364,109 @@ def infer_sex(
-    :param float xx_y_ploidy_cutoff: Max y ploidy for females
-    :param Tuple[float, float] xy_y_ploidy_cutoffs: Boundaries around the chom Y ploidy for males
-    :param float yy_y_ploidy_cutoff: Min chrom Y ploidy for YY
-    :param float xxx_x_ploidy_cutoff: Min chrom X ploidy for XXX
-    :param float f_stat_female_cutoff: Min f-stat for females
-    :param flat f_stat_male_cutoff: Max f-stat for males
-    :return: Struct expression with sex anotations
+    :param NumericExpression f_stat: chrom X f-stat
+    :param float f_stat_cutoff: f-stat to roughly divide 'XX' from 'XY' samples. Assumes XX samples are below cutoff and XY are above cutoff.
+    :param int lower_SD_cutoff: Number of standard deviations to use when determining sex chromosome ploidy cutoffs
+    :param int upper_SD_cutoff: Permissive number of standard deviations to use when determining aneuploidy cutoffs
     :rtype: StructExpression
     """
+    def get_ploidy_cutoffs(ht: hl.Table, f_stat_cutoff: float, lower_SD_cutoff: int, upper_SD_cutoff: int) -> dict:
+        """
+        Gets chromosome X and Y ploidy cutoffs for XY and XX samples
+
+        :param Table ht: Table with samples and their sex chromosome ploidies
+        :param str sex: Sex for which to check X/Y ploidies. Assumes input is either XX or XY.
+        :param float f_stat_cutoff: f-stat cutoff value. Assumes XX values will be below cutoff and XY values will be above cutoff.
+        :param int lower_SD_cutoff: Number of standard deviations to use when determining upper limit of sex chromosome ploidy cutoffs
+        :param int upper_SD_cutoff: Permissive number of standard deviations to use when determining lower ploidy limit and aneuploidy cutoffs
+        :return: Dictionary of ploidy cutoffs; key: sex chrom, values: [lower cut, upper cut, outlier (aneuploidy) cut]
+        :rtype: dict
+        """
+        xx_ht = ht.filter(ht.f_stat < f_stat_cutoff)
+        xy_ht = ht.filter(ht.f_stat > f_stat_cutoff)
+
+        # Get mean/stdev for chrX/Y ploidies
+        xx_x_stats = xx_ht.aggregate(hl.agg.stats(xx_ht.chrX_ploidy))
+        xy_x_stats = xy_ht.aggregate(hl.agg.stats(xy_ht.chrX_ploidy))
+        xx_y_stats = xx_ht.aggregate(hl.agg.stats(xx_ht.chrY_ploidy))
+        xy_y_stats = xy_ht.aggregate(hl.agg.stats(xy_ht.chrY_ploidy))
+
+        # Set cutoffs -- keep cutoffs permissive to account for loss of sex chromosomes
+        cutoffs = {}
+        cutoffs["X"] = [
+                        (xy_x_stats.mean - (upper_SD_cutoff * xy_x_stats.stdev), xy_x_stats.mean + (lower_SD_cutoff * xy_x_stats.stdev)),
+                        (xx_x_stats.mean - (lower_SD_cutoff * xx_x_stats.stdev), xx_x_stats.mean + (lower_SD_cutoff * xx_x_stats.stdev)),
+                        xx_x_stats.mean + (upper_SD_cutoff * xx_x_stats.stdev)
+                        ]
+        cutoffs["Y"] = [
+                        abs(xx_y_stats.mean - (upper_SD_cutoff * xx_y_stats.stdev)),
```

Comment:
I added this `abs` because i wanted to use the upper cutoff of Y ploidy in XX samples as the lower cutoff for a single copy of Y (any samples above this cutoff will get marked as at least having one copy of Y). Unfortunately, I realized when testing the code that Y ploidy distribution in the XX samples was so tight that this value was negative (and in this case, `max(0, ...)` wouldn't work, as every sample would get called as having at least one copy of Y). Happy to change this, let me know what you think!